vrlatency/analysis.py

```diff
@@ -104,59 +104,99 @@ def get_display_dataframe(filename):
     return df
 
 
+def compute_sse(x1, x2, win=100):
+    x1_mat = np.ndarray(buffer=x1, shape=(len(x1)-win, win), strides=(8, 8), dtype=x1.dtype)  # Rolling backwards
+    return np.sum((x1_mat.T - x2[win//2:-win//2]) ** 2, axis=1)
+
+
+def find_global_minimum(x):
+    dx, ddx = np.diff(x), np.diff(x, 2)
+    is_zerocrossing = (dx[1:] * dx[:-1]) < 0
+    is_positive_slope = ddx > 0
+    is_local_minimum = is_zerocrossing & is_positive_slope
+
+    local_minimum_indices = np.where(is_local_minimum)[0] + 1
+    global_minimum_indices = local_minimum_indices[np.argmin(x[local_minimum_indices])]
+    global_minimum_index = int(global_minimum_indices)
+    return global_minimum_index
+
+
+def shift_by_sse(dd):
+
+    sampling_rate = np.diff(dd.TrialTime.values[:2])[0]
+    query = '(-5 < TrialTransitionTime) & (TrialTransitionTime < 5)'
+    dd2 = dd.query(query)
+
+    ref_trial = dd2[dd2.DisplayLatency == dd2.DisplayLatency.min()]  # Min latency used as reference
+    ref_sensor = ref_trial['SensorBrightness'].values
+
+    winsize = 30
+    for trialnum, trial in dd2.groupby('Trial'):
+        test_sensor = trial['SensorBrightness'].values
+        residuals = compute_sse(test_sensor, ref_sensor, win=winsize)
+        minimum = find_global_minimum(residuals)
+        offset = minimum - winsize // 2
+        dd.loc[dd.Trial == trialnum, 'TrialTransitionTime'] -= offset * sampling_rate
+
+    return dd
+
+
 def display_brightness_figure(filename, ax1=None, ax2=None):
 
+    want_to_draw = not bool(ax1 and ax2)
```

Comment:
want_to_draw is a strange variable name for a figure-drawing function.  It points to the function trying to do too much.